### DISCUSSION TITLE
Navigation Server should have a way to state it's ready for requests.

### DISCUSSION THREAD

**Discussion by MichaelJBrune** [godotengine/godot-proposals#6343] [Engine Core]
So I wanted to talk about this before submitting a proposal but I feel like the new navigation servers should have a way to tell users of the navigation server that it's ready for queries. Queries before the navigation server is ready should result in a special NavNotReadyError. Rather than just returning empty arrays for paths or 0,0,0 (Which I might add is a valid location!) for closest point queries.

I feel like this is fairly important and should be created before 4.0 ships otherwise we'd get a lot of confusion with folks using the nav server before it's fully generated correct data.

**Comment by Calinou** [comment]
> I feel like this is fairly important and should be created before 4.0 ships otherwise we'd get a lot of confusion with folks using the nav server before it's fully generated correct data.

4.0 is in release candidate stage, so any new features will be for 4.1 at the earliest.

**Reply by MichaelJBrune** [reply]
Agreed but this isn't a feature request, imho. It's essentially a bug fix.
